docs/cloud-messaging/receive.md

```diff
@@ -409,3 +409,97 @@ experimentalSetDeliveryMetricsExportedToBigQueryEnabled(messaging, true);
+
+### Step 2 - Add target to the Podfile
+
+Ensure that your new extension has access to Firebase/Messaging pod by adding it in the Podfile:
+
+- From the Navigator open the Podfile: Pods > Podfile
+- Scroll down to the bottom of the file and add:
+
+```ruby
+target 'ImageNotification' do
+  use_frameworks!
+  pod 'Firebase/Auth' # Add this line if you are using FirebaseAuth phone authentication 
+  pod 'Firebase/Messaging'
+end
+```
+
+- Install or update your pods using pod install from the ios and/or macos directory.
+
+### Step 3 - Use the extension helper
+
+At this point everything should still be running normally. This is the final step which is invoking the extension helper.
+
+- From the navigator select your ImageNotification extension
+- Open the `NotificationService.m` file
+- At the top of the file import `FirebaseMessaging.h` right after the `NotificationService.h` as shown below:
+
+Replace the content of `NotificationService.m` with
+
+```Objective-C
+#import "NotificationService.h"
+#import "FirebaseMessaging.h"
+#import "FirebaseAuth.h" // Add this line if you are using FirebaseAuth phone authentication
+#import <UIKit/UIKit.h> // Add this line if you are using FirebaseAuth phone authentication
+
+@interface NotificationService ()
+
+@property (nonatomic, strong) void (^contentHandler)(UNNotificationContent *contentToDeliver);
+@property (nonatomic, strong) UNMutableNotificationContent *bestAttemptContent;
+
+@end
+
+@implementation NotificationService
+
+/* Uncomment this if you are using Firebase Auth
+- (BOOL)application:(UIApplication *)app
+            openURL:(NSURL *)url
+            options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options {
+  if ([[FIRAuth auth] canHandleURL:url]) {
+    return YES;
+  }
+  return NO;
+}
+
+- (void)scene:(UIScene *)scene openURLContexts:(NSSet<UIOpenURLContext *> *)URLContexts {
+  for (UIOpenURLContext *urlContext in URLContexts) {
+    [FIRAuth.auth canHandleURL:urlContext.URL];
+  }
+}
+*/
+
+- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
+    self.contentHandler = contentHandler;
+    self.bestAttemptContent = [request.content mutableCopy];
+
+    // Modify the notification content here...
+    [[FIRMessaging extensionHelper] populateNotificationContent:self.bestAttemptContent withContentHandler:contentHandler];
+}
+
+- (void)serviceExtensionTimeWillExpire {
+    // Called just before the extension will be terminated by the system.
+    // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
+    self.contentHandler(self.bestAttemptContent);
+}
+
+@end
+```
+
+### Step 4 - Add the image to the payload
+
+Your device can now display images in a notification by specifying the `imageUrl` option in your FCM payload. Keep in mind that a 300KB max image size is enforced by the device.
```

Comment:
This sentence as written suggest that "Your device" can specify options in the payload.  Suggest rewording to something like "In your notification payload, set <c>."

Also, is `imageURL` an APNs option?  The vanilla notification key is just `image` per this (possibly outdated) doc: https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#notification